llvm/lib/Target/AMDGPU/SIRegisterInfo.cpp

```diff
@@ -1118,9 +1118,7 @@ SIRegisterInfo::getPointerRegClass(unsigned Kind) const {
 
 const TargetRegisterClass *
 SIRegisterInfo::getCrossCopyRegClass(const TargetRegisterClass *RC) const {
-  if (RC == &AMDGPU::SCC_CLASSRegClass)
-    return getWaveMaskRegClass();
-  return RC;
+  return RC == &AMDGPU::SCC_CLASSRegClass ? &AMDGPU::SReg_32RegClass : RC;
```

Comment:
Shouldn't SCC_CLASS depend on the wavesize?

Reply:
No. These have nothing to do with each other. To extract a value into an allocatable register, a 32-bit SGPR is the natural choice